### PR TITLE
ENH+BF: make it possible to export_to_figshare without cd'ing

### PR DESCRIPTION
**Pull request by yarikoptic**
Done by allowing export-archive --filename=directorypath/ so we could
point to the directory where the tarball could be located

### CODE REVIEW

File: datalad/plugin/export_archive.py [hunk 1]
```diff
@@ -106,8 +108,11 @@ def _filter_tarinfo(ti):
                 '.' if compression else '',
                 compression) if archivetype == 'tar' else '')
 
+        filename_tmpl = "datalad_{.id}".format(dataset)
```
[the remark this answers is not in the document]
Comment (yarikoptic) [reply datalad/plugin/export_archive.py]: yeah, indeed... I guess I had some visions of making it into cmdline option template so we could provide dataset etc objects for it to get filled in.   renaming to `default_filename`

File: datalad/plugin/export_to_figshare.py [hunk 2]
```diff
@@ -230,7 +230,11 @@ def __call__(dataset, filename=None, missing_content='error', no_annex=False,
             raise RuntimeError(
                 "Paranoid authors of DataLad refuse to proceed in a dirty repository"
             )
-        lgr.info("Exporting current tree as an archive since figshare does not support directories")
+        lgr.info(
+            "Exporting current tree as an archive since figshare "
+            "does not support directories")
```
[the remark this answers is not in the document]
Comment (yarikoptic) [reply datalad/plugin/export_to_figshare.py]: ok, adding, thanks

File: datalad/plugin/export_to_figshare.py [hunk 3]
```diff
@@ -165,8 +165,8 @@ class ExportToFigshare(Interface):
             metavar="PATH",
             nargs='?',
             doc="""File name of the generated ZIP archive. If no file name is
-            given the archive will be generated in the current directory and
-            will be named: datalad_<dataset_uuid>.zip.""",
+            given the archive will be generated in the top directory
+            of the dataset and will be named: datalad_<dataset_uuid>.zip.""",
```
[the remark this answers is not in the document]
Comment (yarikoptic) [reply datalad/plugin/export_to_figshare.py]: `export_archive` is typically used to "export" it to the drive outside of the dataset e.g. for archival or sharing with a buddy, and we would use it with `-d` or just as  a method of a dataset if under python.  We wouldn't want it typically to be generated under dataset directory IMHO.
figshare and alike, where we need to upload it and desire to addurl it back, so it should be exported under dataset is atypical imho for plain `export_archive` usecase.

**Comment by kyleam** [comment]
I'll merge this.  There's no coverage on the export_to_figshare bits, but we're no worse off since that file doesn't have any coverage to begin with.

**Pull request merged**